backend/src/storage/blob.rs

```diff
@@ -50,59 +155,301 @@ impl StorageProvider for BlobStorageProvider {
             .map_err(|e| {
                 StorageError::DatabaseError(format!("failed to convert to utf8: {}", e))
             })?;
-        println!("s_content == {s_content}");
 
-        let flag: BoolFlag = serde_json::from_str(&s_content)
-            .map_err(|e| StorageError::DatabaseError(format!("failed to parse json: {}", e)))?;
+        let flag: BoolFlag = serde_json::from_str(&s_content).map_err(|e| {
+            StorageError::SerializationError(format!("failed to parse json: {}", e))
+        })?;
 
         Ok(Some(flag))
     }
 
-    async fn create_bool_flag(&self, _flag: BoolFlag) -> Result<(), StorageError> {
-        todo!()
+    async fn create_bool_flag(&self, flag: BoolFlag) -> Result<(), StorageError> {
+        if self.get_bool_flag(&flag.id).await?.is_some() {
+            return Err(StorageError::AlreadyExists);
+        }
+        if self.name_exists(&flag.name).await? {
+            return Err(StorageError::AlreadyExists);
+        }
+
+        let blob_client = self.container_client.blob_client(flag.id.clone());
+        
+        let flag_str = serde_json::to_string(&flag).map_err(|e| {
+            StorageError::SerializationError(format!("failed to serialize json: {}", e))
+        })?;
+
+        let _res = blob_client
```

Comment:
looks like we don't do anything with `_res`, can we remove the binding ?

Reply:
yes will adjust